walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,90 @@
 
 package com.google.sps;
 
+import java.util.Queue;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Comparator;
+import java.util.ArrayList;
+import java.util.List;
+
+
+// A meeting request has:
+// a name
+// a duration in minutes
+// a collection of attendees
+
+// For a particular time slot to work, all attendees must be free to attend the meeting. When a query is made, it will be given a collection of all known events. Each event has:
+// a name
+// a time range
+// a collection of attendees
+// A time range will give you the start time, the end time, and the duration of the event. If you want to know more, open the TimeRange.java file.
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      //Gets info on request
+      Collection<String> attendees = request.getAttendees();
+      long dur = request.getDuration();
+      if(dur > TimeRange.END_OF_DAY - TimeRange.START_OF_DAY || dur < 0){
+          return new ArrayList<>(); //Special case where there can be no meetings of that duration 
+      }
+      //Find busy times of each person attending event
+      List<TimeRange> busyTimes = new ArrayList<>();
+      for(Event e : events){ //looks at every event
+          Collection<String> eventAtt = e.getAttendees();
+          for(String a : eventAtt){ //looks at all the people attending the event
+                System.out.println(a + eventAtt.size() + " " + attendees.contains(a));
+              if(attendees.contains(a)){
+                  System.out.println("Goes in");
+                  busyTimes.add(e.getWhen());
+                  break; //found someone in this event, we can stop searching this one
+              }
+          }
+
+      }
+      List<TimeRange> tr = new ArrayList(); 
```

Comment:
Same here about variable names "validTimeRanges" is a bit easier to understand here.